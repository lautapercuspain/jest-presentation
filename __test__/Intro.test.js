/**
 * Sample React Native Snapshot Test
 */
'use strict';
import React from 'react';
import Intro from '../Intro';
import { ActivityIndicator,
         Image, 
         TextInput, 
         ListView,
         Text,} from 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('It renders a group of tests', () => {

    it('renders the Intro component', () => {
        const tree = renderer.create(<Intro />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // These serve as integration tests for the jest-react-native preset.
    it('renders the ActivityIndicator component', () => {
        const tree = renderer.create(<ActivityIndicator animating={false} size="small" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the Image component', done => {
       
        Image.getSize('path.jpg', (width, height) => {
            const tree = renderer.create(<Image style={{ height, width }} />).toJSON();
            expect(tree).toMatchSnapshot();
            done();
        });
    });

    it('renders the TextInput component', () => {
        const tree = renderer.create(<TextInput autoCorrect={false} value="apple banana kiwi" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders the ListView component', () => {
        const dataSource = new ListView.DataSource({ 
            rowHasChanged: (r1, r2) => r1 !== r2,
        }).cloneWithRows(['apple', 'banana', 'kiwi']);
        const tree = renderer
            .create(<ListView pagingEnabled={true} dataSource={dataSource} renderRow={rowData => <Text>{rowData}</Text>} />)
            .toJSON();
  
        expect(tree).toMatchSnapshot(); 
    });
});
