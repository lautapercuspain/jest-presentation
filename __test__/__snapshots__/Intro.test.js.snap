// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders a group of tests renders the ActivityIndicator component 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
/>
`;

exports[`It renders a group of tests renders the Image component 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`It renders a group of tests renders the Intro component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "margin": 10,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    This is a React Native snapshot test.
  </Text>
</View>
`;

exports[`It renders a group of tests renders the ListView component 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 3,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  pagingEnabled={true}
  removeClippedSubviews={true}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}
  stickySectionHeadersEnabled={true}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    apple
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    banana
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    kiwi
  </Text>
</ScrollView>
`;

exports[`It renders a group of tests renders the TextInput component 1`] = `
<TextInput
  autoCorrect={false}
  value="apple banana kiwi"
/>
`;
